File: app/src/main/java/hu/epam/worktime/mvvmworkdroid/common/widgets/recyclerview/ListItemViewModel.java
package hu.epam.worktime.mvvmworkdroid.common.widgets.recyclerview;

import android.databinding.BaseObservable;

/**
 * Abstract viewmodel for list items which provides the viewtype of the item.
 * Created by Mate_Redecsi on 10/21/2016.
 */
public abstract class ListItemViewModel extends BaseObservable {
    public abstract int getViewType();
}

File: app/src/main/java/hu/epam/worktime/mvvmworkdroid/common/widgets/recyclerview/ListItemViewModel.kt
package hu.epam.worktime.mvvmworkdroid.common.widgets.recyclerview

import android.databinding.BaseObservable

/**
 * Abstract viewmodel for list items which provides the viewtype of the item.
 * Created by Mate_Redecsi on 10/21/2016.
 */
abstract class ListItemViewModel : BaseObservable() {
    abstract val viewType: Int
}
